docs/utils/threading/worker-pool.md

```diff
@@ -148,3 +148,129 @@ functionality in Thunder, which will be further described in a different documen
+            else {
+                _threadPool.Submit(job, Core::infinite);
+            }
+        }
+```
+
+
+#### Reschedule
+`Reschedule()` method is used to change the execution date of an already scheduled `job`. It is used in the same way as the `Schedule()` method. It works as follows, first the scheduled `job` is attempted to be canceled and then the Schedule with the newly specified time is used.
+!!! note
+    `Reschedule()` returns true when the `job` cancellation succeeds and false when it fails, this can happen when the given task was not previously scheduled using `Schedule()`
+
+```cpp
+bool Reschedule(const Core::Time& time, const Core::ProxyType<IDispatch>& job) override
+        {
+            ASSERT(job.IsValid() == true);
+
+            bool revoked = (Revoke(job) != Core::ERROR_UNKNOWN_KEY);
+            Schedule(time, job);
+            return (revoked);
+        }
+```
+
+
+#### Revoke
+`Revoke()` is used to cancel the execution of a `job`. We use it by specifying the `job` we want to cancel and the time after which it should be cancelled. We do not need to specify the time, if we leave the default value, the `job` will be canceled immediately. The `Reschedule` method returns a numeric value.
+
+ Error codes that can be returned:
+
+* `ERROR_NONE` - 0 (no errors)
+* `ERROR_UNKNOWN_KEY` - 22
+
+!!!warning
+    `ERROR_UNKNOWN_KEY` will be returned in case the job was not previously scheduled!
+
+```cpp
+uint32_t Revoke(const Core::ProxyType<IDispatch>& job, const uint32_t waitTime = Core::infinite) override
+        {
+            uint32_t result(_timer.Revoke(Timer(this, job)) ? Core::ERROR_NONE : Core::ERROR_UNKNOWN_KEY);
+
+            uint32_t report = _threadPool.Revoke(job, waitTime);
+
+            if (report == Core::ERROR_UNKNOWN_KEY) {
+                report = _external.Completed(job, waitTime);
+
+                if ( (report != Core::ERROR_UNKNOWN_KEY) && (result == Core::ERROR_UNKNOWN_KEY) ) {
+                    result = report;
+                }
+            }
+
+            return (result);
+        }
+```
+
+### Example
+`Scheduler` is used for example in rdkservices repository. Here is example use of `Schedule()` from there.
+```cpp
+void CFrontPanel::startBlinkTimer(int numberOfBlinkRepeats)
+        {
+            LOGWARN("startBlinkTimer numberOfBlinkRepeats: %d m_blinkList.length : %zu", numberOfBlinkRepeats, m_blinkList.size());
+            stopBlinkTimer();
+            m_numberOfBlinks = 0;
+            m_isBlinking = true;
+            m_maxNumberOfBlinkRepeats = numberOfBlinkRepeats;
+            m_currentBlinkListIndex = 0;
+            if (m_blinkList.size() > 0)
+            {
+                FrontPanelBlinkInfo blinkInfo = m_blinkList.at(0);
+                setBlinkLed(blinkInfo);
+                if (m_isBlinking)
+                    blinkTimer.Schedule(Core::Time::Now().Add(blinkInfo.durationInMs), m_blinkTimer);
+            }
+        }
+```
+
+Below is test example of using it.
```

Comment:
I think it would be better to find actual examples in code where it is used, it must be used at least a few times in  https://github.com/rdkcentral/ThunderNanoServices, and then we you give an example from an actual code, you can write about why it was used here and how it is actually useful in practice and not just in theory.